### PR TITLE
Azure stack provisioning

### PR DESCRIPTION
**Pull request by sshveta**
{{pytest: -v --long-running --use-provider azure cfme/tests/services/test_provision_stack.py }}

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit 95ed748b82e24cdf6944c1d47adac82538b6e713:

Everything seems all right :smile: .
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit eadf5f0fc380d47c54b4e8383d5843e4bd536583:

Everything seems all right from lint perspective :smile: .

Commit inspection report:
**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 7efde1c6c47c561b3a65d40c968c1eefd4511d6d:

Everything seems all right from lint perspective :smile: .

Commit inspection report:
**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit c122a70997faa5560b39358f0efedeb4a79500ac:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
<mergeable/>

*CFME QE Bot*

**Comment by sshveta** [comment]
@Jeff : PR fails because azure account is missing resources. Service request fails with error with this account. 
Checked with azurejt account , request passes.

File: cfme/tests/services/test_provision_stack.py [hunk 1]
```diff
@@ -140,10 +140,15 @@ def pytest_generate_tests(metafunc):
 
 
 @pytest.fixture(scope="function")
-def dialog(provisioning):
-    dialog_name = "dialog_" + fauxfactory.gen_alphanumeric()
+def dialog(provider, provisioning):
     template_type = provisioning['stack_provisioning']['template_type']
-    template = OrchestrationTemplate(template_type=template_type,
+    if provider.type == "azure":
```
Comment (psav) [inline cfme/tests/services/test_provision_stack.py]: Why is the template name the only thing that changes if it is azure? Is this conditional needed?

File: cfme/tests/services/test_provision_stack.py [hunk 2]
```diff
@@ -285,8 +340,23 @@ def test_retire_stack(provider, create_template, catalog, request):
                   dialog=dialog_name, orch_template=template.template_name,
                   provider_type=provider.name)
     catalog_item.create()
-    stackname = "test" + fauxfactory.gen_alphanumeric()
-    stack_data = {'stack_name': stackname}
+    if provider.type == "azure":
```
Comment (psav) [inline cfme/tests/services/test_provision_stack.py]: Seeing as the data seems to be cropping up everywhere, I propose that we DRY it out into a function, something like ```prepare_stack_data()``` which takes ```stackname``` as a required param, and then optionally other params and the provider object. In fact, thinking about it, it looks like stack_data could be a fixture as the contents look identical, then the logic of deciding on the stack_data is not inside the test, it is a lot easier to read.

File: cfme/tests/services/test_provision_stack.py (hunk 1 above)
Comment (sshveta) [inline cfme/tests/services/test_provision_stack.py]: Azure comes with a pre built template and dialog. so Using that in the test. 
Where as for other providers , we need to add a new template

**Comment by dajoRH** [comment]
Lint report for commit 6dd871a11122afe1e1c8cfff1a1b363385d263c1:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
<mergeable/>

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit dbce75485f57185b73cdd260dd41a8efb2055c3d:

`cfme/services/catalogs/service_catalogs.py`:
- :red_circle: Line 9:1: **F401** *'time' imported but unused*

Please, rectify these issues :smirk: .

Commit inspection report:

**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 2cd6d82fc0f2b478e86835a177a006c4fbdae66b:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
<mergeable/>

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit bb88a5ef08ebb9a408fd58d70776a5545599a1db:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
<mergeable/>

*CFME QE Bot*

File: cfme/tests/services/test_provision_stack.py (hunk 1 above)
Comment (psav) [inline cfme/tests/services/test_provision_stack.py]: OK, no worries

**Comment by psav** [comment]
Code looks good, just need to see PRT pass a bit

**Comment by dajoRH** [comment]
Lint report for commit 5f4ed01dba6fbb9345c81e25660b6f7a681c7b01:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
<mergeable/>

*CFME QE Bot*

**Pull request merged**